src/models/data.h

```diff
@@ -46,23 +46,40 @@ QString prettifySymbol(const QString& symbol);
 
 struct Symbol
 {
+    static bool prettifySymbol;
+
     Symbol(const QString& symbol = {}, const QString& binary = {}, const QString& path = {})
         : symbol(symbol)
-        , prettySymbol(prettifySymbol(symbol))
         , binary(binary)
         , path(path)
     {
     }
 
     // function name
     QString symbol;
-    // prettified function name
-    QString prettySymbol;
+    // prettified function name (cached in getSymbol)
+    mutable QString cachedPrettySymbol;
     // dso / executable name
     QString binary;
     // path to dso / executable
     QString path;
 
+    QString getSymbol() const
```

Comment:
please remove this logic here and instead add a free function helper to Util that takes a `const Symbol&` input and then checks the settings singleton and returns either the pretty symbol or the un-prettified symbol